legacy/c3_trajectory_generator/src/waypoint_validity.cpp

```diff
@@ -0,0 +1,61 @@
+#include "waypoint_validity.hpp"
+
+// Point and sub_size must be relative to ogrid, IE: meters * 1/OGRID_RESOLUTION + offset
+bool WaypointValidity::check_if_hit(cv::Point center, cv::Size sub_size)
+{
+  for (int x = center.x - sub_size.width / 2; x < center.x + sub_size.width / 2; ++x)
+  {
+    for (int y = center.y - sub_size.height / 2; y < center.y + sub_size.height / 2; ++y)
+    {
+      if (ogrid_map_->data.at(x + y * ogrid_map_->info.width) == 99)
```

Comment:
Cast the value of the enum into an int here for the comparison. You don't want this to break if you change the enum.